packages/@uppy/react/src/Dashboard.ts

```diff
@@ -0,0 +1,109 @@
+import { createElement as h, Component } from 'react'
+import PropTypes from 'prop-types'
+import type { UnknownPlugin, Uppy } from '@uppy/core'
+import DashboardPlugin from '@uppy/dashboard'
+import type { Body, Meta } from '@uppy/utils/lib/UppyFile'
+import type { DashboardOptions } from '@uppy/dashboard'
+import {
+  locale,
+  uppy as uppyPropType,
+  plugins,
+  metaFields,
+  cssSize,
+} from './propTypes.ts'
+import getHTMLProps from './getHTMLProps.ts'
+import nonHtmlPropsHaveChanged from './nonHtmlPropsHaveChanged.ts'
+
+type DashboardInlineOptions<M extends Meta, B extends Body> = Omit<
+  DashboardOptions<M, B> & { inline: true },
+  'inline'
+> &
+  React.BaseHTMLAttributes<HTMLDivElement>
```

Comment:
Should have reviewed this in the other PR but didn't foresee how we'd use it, but I initially envisioned we'd import `DashboardInlineOptions` which does this trick inside `@uppy/dashboard`. Then we keep the conditions of the discriminated union in one place instead of repeating it everywhere.

What do you think? Doesn't need a separate PR.